File: core/src/main/java/de/qabel/core/repository/sqlite/DesktopClientDatabase.java
package de.qabel.core.repository.sqlite;

import java.sql.Connection;

import de.qabel.core.repository.sqlite.migration.*;

public class DesktopClientDatabase extends AbstractClientDatabase {

    public DesktopClientDatabase(Connection connection) {
        super(connection);
    }

    public AbstractMigration[] getMigrations(Connection connection) {
        return new AbstractMigration[]{
            new Migration1460367000CreateIdentitiy(connection),
            new Migration1460367005CreateContact(connection),
            new Migration1460367010CreateAccount(connection),
            new Migration1460367015ClientConfiguration(connection),
            new Migration1460367020DropState(connection),
            new Migration1460367025BoxSync(connection),
            new Migration1460367030ShareNotification(connection),
            new Migration1460367035Entity(connection),
            new Migration1460367040DropMessage(connection),
            new Migration1460987825PreventDuplicateContacts(connection),
            new Migration1460997040ChatDropMessage(connection),
            new Migration1460997041RenameDropState(connection)
        };
    }

}

File: core/src/main/java/de/qabel/core/repository/sqlite/DesktopClientDatabase.kt
package de.qabel.core.repository.sqlite

import de.qabel.core.repository.sqlite.migration.*
import java.sql.Connection

open class DesktopClientDatabase(connection: Connection) : AbstractClientDatabase(connection) {

    override var version by PragmaVersionAdapter(connection)

    override fun getMigrations(connection: Connection): Array<AbstractMigration> {
        return arrayOf(Migration1460367000CreateIdentitiy(connection),
            Migration1460367005CreateContact(connection),
            Migration1460367010CreateAccount(connection),
            Migration1460367015ClientConfiguration(connection),
            Migration1460367020DropState(connection),
            Migration1460367025BoxSync(connection),
            Migration1460367030ShareNotification(connection),
            Migration1460367035Entity(connection),
            Migration1460367040DropMessage(connection),
            Migration1460987825PreventDuplicateContacts(connection),
            Migration1460997041RenameDropState(connection),
            Migration1460997042ExtendContact(connection),
            Migration1460997043ContactDefaults(connection),
            Migration1460997044IndexIdentity(connection),
            Migration1460997045UploadableIdentity(connection),
            Migration1476108744PrefixTypes(connection)
        )
    }
}
